### DISCUSSION TITLE
[Question] ASR Model Neural Rescoring

### DISCUSSION THREAD

**Discussion by jyseol-inno** [NVIDIA-NeMo/NeMo#4458] [Q&A] [answered]
Hello.
I am currently studying the Korean asr model and testing it.
I learned the transformer_lm model [https://github.com/NVIDIA/NeMo/blob/stable/examples/nlp/language_modeling/transformer_lm.py](https://github.com/NVIDIA/NeMo/blob/stable/examples/nlp/language_modeling/transformer_lm.py) and combined it with the asr model to get the following results. [https://github.com/NVIDIA/NeMo/blob/stable/scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py](https://github.com/NVIDIA/NeMo/blob/stable/scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py)
Does the 'Best possible WER' score mean the score that can be obtained through hyperparameter tuning?

![image](https://user-images.githubusercontent.com/100547581/176077899-4d9790d2-4837-434b-8aa5-d6f5a07c930f.png)

**Comment by titu1994** [comment, marked as answer]
We should really rename that. "Best" here is Oracle scores - ie given a perfect LM what can the AM attain. It is not the realistic result because it needs ground truth to compute this Oracle score. 

Your actual result is the LM rescoring WER (CER) above that.

**Reply by titu1994** [reply]
Fyi @VahidooX - let's remove "Best" and write "Oracle" instead

**Reply by VahidooX** [reply]
@titu1994 I would change it in a PR.
@jyseol-inno Have you used N-gram LM to get the initial predictions or just beam search decoding?
How many samples did you use to train the transformer model?

The best here is not about hyperparameters but shows what you could achieve if you had an oracle LM to select the correct one among the candidates.

**Reply by jyseol-inno** [reply]
@VahidooX 
There are scores combined with beam search decoding and 5-gram NM.
The model trained with 1793996 samples from the Korean speech recognition dataset.
![image](https://user-images.githubusercontent.com/100547581/176567122-0019fad6-0985-46ee-b11e-b8afd1b48873.png)

**Reply by titu1994** [reply]
Why is beam search causing your results to get worse ?
